src/test/java/jcifs/tests/SymLinkTest.java

```diff
@@ -0,0 +1,188 @@
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+package jcifs.tests;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Properties;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jcifs.CIFSException;
+import jcifs.DialectVersion;
+import jcifs.config.PropertyConfiguration;
+import jcifs.context.BaseContext;
+import jcifs.smb.NtlmPasswordAuthenticator;
+import jcifs.smb.SmbException;
+import jcifs.smb.SmbFile;
+
+
+/**
+ * The actual SMB share must be configured in the system properties
+ * with the 'real' server settings for testing.
+ *
+ * You must create the following folder structures in your SMB share
+ * for a successful completion of these unit tests:
+ *
+ * <pre class="code">
+ *  smbSource/
+ *  |-- subSmbSource/
+ *      |-- subSmbSource1.txt - contains 'subSource1'
+ *      |-- subSmbSource2.txt - contains 'subSource2'
+ *  smbTarget/
+ *  |-- foo.file - contains 'foofoofoo'
+ *  symlinkTestDir/ - symbolic link to directory smbSource/subSmbSource/
+ *                    using 'mklink /D symlinkTestDir <share root>\smbSource\subSmbSource\'
+ *  symlinkTestFoo.file - symbolic link to file smbTarget/foo.file
+ *                        using 'mklink symlinkTestFoo.file <share root>\smbTarget\foo.file'
+ * </pre>
+ *
+ * @author Gregory Bragg
+ */
+@RunWith ( Parameterized.class )
+@SuppressWarnings ( "javadoc" )
+public class SymLinkTest extends BaseCIFSTest {
+
+    private static final Logger log = LoggerFactory.getLogger(SymLinkTest.class);
+
+    private static final String SMB_FILE_SEPARATOR = "/";
+
+    public SymLinkTest ( String name, Map<String, String> properties ) {
+        super(name, properties);
+    }
+
+
+    @Parameters ( name = "{0}" )
+    public static Collection<Object> configs () {
+        return getConfigs("smb1", "noUnicode", "forceUnicode", "noNTStatus", "noNTSmbs", "smb2", "smb30", "smb31");
```

Comment:
As symlinks are only supported with SMB 3.1.1 I don't think there is a need to run this against SMB1 and other SMB2 versions.

Reply:
Admittedly I do not fully understand how you have your JUnit tests configured... I'll attempt to make a change.

Reply:
Don't worry, I can take care of this when merging.